agent-inject/agent/annotations.go

```diff
@@ -862,6 +862,14 @@ func (a *Agent) telemetryConfig() map[string]interface{} {
 func (a *Agent) authConfig() map[string]interface{} {
 	authConfig := make(map[string]interface{})
 
+	if a.Vault.Role != "" {
+		authConfig["role"] = a.Vault.Role
+	}
+
+	if a.ServiceAccountTokenVolume.MountPath != "" && a.ServiceAccountTokenVolume.TokenPath != "" {
```

Comment:
It would be good to add a comment as to why this is being set before the auth-config loop (namely that the `serviceaccount()` function may have picked the first serviceaccount volume it finds.